wallet/src/de/schildbach/wallet/ui/main/MainViewModel.kt

```diff
@@ -379,16 +381,26 @@ class MainViewModel @Inject constructor(
         }
     }
 
-    suspend fun getDeviceTimeSkew(): Long {
+    suspend fun getCoinJoinMode(): CoinJoinMode {
+        return coinJoinConfig.getMode()
+    }
+
+    suspend fun getDeviceTimeSkew(): Pair<Boolean, Long> {
         return try {
             val systemTimeMillis = System.currentTimeMillis()
             val result = HTTP_CLIENT.head("https://www.dash.org/")
             val networkTime = result.headers.getDate("date")?.time
             requireNotNull(networkTime)
-            abs(systemTimeMillis - networkTime)
+            val maxAllowedTimeSkew = if (coinJoinConfig.getMode() == CoinJoinMode.NONE) {
+                TIME_SKEW_TOLERANCE
+            } else {
+                TIME_SKEW_TOLERANCE_COINJOIN
+            }
+            val timeSkew = abs(systemTimeMillis - networkTime) / 1000
+            return Pair(timeSkew > maxAllowedTimeSkew, timeSkew)
```

Comment:
for the MainActivity we will show one message if CoinJoin is ON, and another if OFF.  Each has different tolerances.